recipes/zlib/1.2.11/conanfile.py

```diff
@@ -27,24 +34,15 @@ def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
+        if self.options.minizip != "deprecated":
+            self.output.warn("minizip option is deprecated. Please use the new minizip/1.2.11 package")
+
+    def package_id(self):
+        del self.info.options.minizip
+
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
-
-    def build(self):
-        self._build_zlib()
-        if self.options.minizip:
-            self._build_minizip()
-
-    def _build_zlib_cmake(self):
-        cmake = CMake(self)
-        cmake.configure(build_dir=".")
-        # we need to build only libraries without test example/example64 and minigzip/minigzip64
-        make_target = "zlib" if self.options.shared else "zlibstatic"
-
-        cmake.build(build_dir=".", target=make_target)
-
-    def _build_zlib(self):
         for patch in self.conan_data["patches"][self.version]:
             tools.patch(**patch)
```

Comment:
patch must be executed on build method, not source.

Reply:
Oups :) I used as guidance the boost recipe and unfortunately the patch phase is in the source function.

Reply:
Boost doesn't copy sources to build folder. It's an exception because that package is huge.